backend/src/endpoints/users/users.controller.ts

```diff
@@ -215,5 +215,39 @@ export default class UsersController {
         res.status(500).describe("unknown server error occurred").send();
       });
   }
+  
+  static updateUserProgress(req: Request, rawRes: Response) {
+    const res = new ApiResponse(rawRes);
+    const logger = UsersController.fileLogger.createFunctionLogger("updateUserScore");
+    const username = req.params.username;
+    let response;
+    const {
+      contentProgress,
+    } = req.body;
+    logger.debug(`request recieved: get user with username ${username}`);
+    if (contentProgress < 10)
```

Comment:
We cannot always be certain that a lesson will contain exactly 10 pieces of content.

Reply:
Passed questionList.length to the api body.